src/lints/ast.rs

```diff
@@ -0,0 +1,359 @@
+use anyhow::Context;
+use pg_query::protobuf::{
+    AlterTableCmd, AlterTableType, ColumnDef, CreateStmt, CreateTableAsStmt, IndexStmt,
+    VariableSetStmt,
+};
+
+/// A simpler, linter-rule friendly representation of the postgres parse tree
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum StatementSummary {
+    Ignored,
+    LockTimeout,
+    CreateTable {
+        schema: String,
+        name: String,
+    },
+    CreateTableAs {
+        schema: String,
+        name: String,
+    },
+    CreateIndex {
+        schema: String,
+        idxname: String,
+        concurrently: bool,
+        target: String,
+    },
+    AlterTable {
+        schema: String,
+        name: String,
+        actions: Vec<AlterTableAction>,
+    },
+}
+
+impl StatementSummary {
+    /// Returns a list of (schema, name) tuples for objects created by this statement
+    pub fn created_objects(&self) -> Vec<(&str, &str)> {
+        match self {
+            StatementSummary::CreateIndex {
+                schema, idxname, ..
+            } => vec![(schema, idxname)],
+            StatementSummary::CreateTable { schema, name } => vec![(schema, name)],
+            StatementSummary::CreateTableAs { schema, name } => vec![(schema, name)],
+            _ => vec![],
```

Comment:
I think it's probably better to explicitly mention the ignored patterns so the compiler will flag it when we add new ones. Wonder if there's a rust lint for that 🤔